v0/destinations/freshsales/transform.js

```diff
@@ -0,0 +1,186 @@
+const { EventType } = require("../../../constants");
+const {
+  defaultRequestConfig,
+  CustomError,
+  constructPayload,
+  ErrorMessage,
+  getErrorRespEvents,
+  getSuccessRespEvents
+} = require("../../util");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require("./config");
+const {
+  getAccountDetails,
+  getUserAccountDetails,
+  checkNumberDataType
+} = require("./utils");
+
+/*
+ * This functions is used for creating response for identify call, to create or update contacts.
+ * @param {*} message
+ * @param {*} Config
+ * @returns
+ */
+const identifyResponseBuilder = (message, { Config }) => {
+  const payload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.IDENTIFY.name]
+  );
+
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+
+  checkNumberDataType(payload);
+  const response = defaultRequestConfig();
+  response.headers = {
+    Authorization: `Token token=${Config.apiKey}`,
+    "Content-Type": "application/json"
+  };
+  response.endpoint = `https://${Config.domain}${CONFIG_CATEGORIES.IDENTIFY.endpoint}`;
+  response.method = CONFIG_CATEGORIES.IDENTIFY.method;
+  response.body.JSON = {
+    contact: payload,
+    unique_identifier: { emails: payload.emails }
+  };
+  return response;
+};
+
+/*
+ * This functions is used for associating contacts in account.
+ * @param {*} message
+ * @param {*} Config
+ * @returns
+ */
+const groupResponseBuilder = async (message, { Config }) => {
+  const payload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.GROUP.name]
+  );
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+
+  checkNumberDataType(payload);
+  const payloadBody = {
+    unique_identifier: { name: payload.name },
+    sales_account: payload
+  };
+
+  const accountResponse = await getAccountDetails(payloadBody, Config);
+
+  const accountId = accountResponse.response.sales_account
+    ? accountResponse.response.sales_account.id
+    : undefined;
+  if (!accountId) {
+    throw new CustomError("Error while fetching accountId");
+  }
+  const userEmail = message.context.traits
+    ? message.context.traits.email
+    : undefined;
+  if (!userEmail) {
+    throw new CustomError("Invalid userEmail");
+  }
+  const userSalesAccountResponse = await getUserAccountDetails(
+    userEmail,
+    Config
+  );
+  let accountDetails = userSalesAccountResponse.response.contact
+    ? userSalesAccountResponse.response.contact.sales_accounts
+    : undefined;
+  if (!accountDetails) {
+    throw new CustomError("Error while fetching user accountDetails");
+  }
+  if (accountDetails.length > 0) {
+    accountDetails = [
+      ...accountDetails,
+      {
+        id: accountId,
+        is_primary: false
+      }
+    ];
+  } else {
+    accountDetails = {
+      id: accountId,
+      is_primary: true
+    };
+  }
```

Comment:
```
const accountDetail = {
        id: accountId,
        is_primary: false
      }
if (accountDetails.length > 0) {
    accountDetails = [
      ...accountDetails,
      accountDetail
    ];
  } else {
    accountDetails = [accountDetail]
  }
```

Reply:
Is it correct? If there is not any accountDetail available for an existing user we have to set this account as primary.